app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -415,8 +416,8 @@ class BrowserTabFragment :
     }
 
     private val autofillCallback = object : Callback {
-        override suspend fun onCredentialsAvailableToInject(credentials: List<LoginCredentials>, isAutoprompt: Boolean) {
-            showAutofillDialogChooseCredentials(credentials, isAutoprompt)
+        override suspend fun onCredentialsAvailableToInject(credentials: List<LoginCredentials>, triggerType: LoginTriggerType) {
+            showAutofillDialogChooseCredentials(credentials, triggerType)
```

Comment:
I'm probably missing context but I don't get why this is needed?

From what I understand this callback is called in the `impl` module and we pass some information here (to the tab fragment) which is now going to call `credentialAutofillDialogFactory.autofillSelectCredentialsDialog` passing the information back `impl` so it can return a `DialogFragment`. 

This roundtrip seems rather unnecessary and is probably easily avoidable if the methods in `CredentialAutofillDialogFactory` returned a `DialogFragment` rather than a specific type that is never needed. In fact, those dialogs implement `DialogFragmentType` which has a method to return a `DialogFragment` and that implementation is a simple `return this`.

Reply:
Hmm, I do agree that this can be simplified. Since I think the logic within `viewModel.returnNoCredentialsWithPage `and  onCredentialsAvailableToSave can all be handled within `impl` (`AutofillStoredBackJavascriptInterface`).

Though I think even if we get the `DialogFragment` ,we would need the `BrowserTabFragment.showDialogHidingPrevious()` to show it since we don't have any reference to the `childFragmentManager` within the autofill feature. So maybe this call back could just be simplified to `DialogFragment -> {}`

Reply:
Yup, 100% the dialog will have to be shown by the tab fragment.